soccer/planning/RRTPlanner.cpp

```diff
@@ -180,47 +184,57 @@ Planning::InterpolatedPath RRTPlanner::makePath()
 	return newPath;
 }
 
-void RRTPlanner::optimize(Planning::InterpolatedPath &path, const Geometry2d::CompositeShape *obstacles, const MotionConstraints &motionConstraints, Geometry2d::Point vi)
+Planning::InterpolatedPath* RRTPlanner::optimize(Planning::InterpolatedPath &path, const Geometry2d::CompositeShape *obstacles, const MotionConstraints &motionConstraints, Geometry2d::Point vi)
```

Comment:
Well its modifying it in place or returning null_ptr if no valid path could be planned.
This class needs to be nuked anyways.